### ISSUE TITLE
q-select applies data-cy multiple times

### ISSUE THREAD

**Issue by Quasarman** [quasarframework/quasar#11375]
The q-select component applies the data-cy attribute not only to the input but also the div containing the actual input element. I am not sure if this is intended or not but it is kind of counter intuitive since when runing cypress it will find 2 elements which will cause issues when trying to cy.type() inside the q-select input.

Possible workarounds include targeting only inputs with the custom data-cy attribute but again this seems like a unnecessary step, considering this does not have to be done for e.g. the q-input component since there the data-cy attribute gets applied only once.

Quasar version: 1.15.18

**Comment by pdanpdan** [comment]
Please fill the issue template and add a codepen, ~because I cannot reproduce it~:
- official branch: https://codepen.io/pdanpdan/pen/VwzqKRN?editors=101
- my testing branch: https://codepen.io/pdanpdan/pen/GRvPjeK?editors=101

I suppose you meant if `use-input` is set.

**Comment by Quasarman** [comment]
Sorry for not being explicit enough I typed it on my phone since I was no longer on my PC yesterday. But now that I am, I had another look into it. I took the first example from the docs: https://v1.quasar.dev/vue-components/select#filtering-and-autocomplete where the "Filtering and Autocomplete" features of the `q-select` component are described. But you are right I suppose the behavior can be observed on any q-select that has the `use-input` property set.

Anyways what I did is I added the `data-cy` prop to the example inside code pen like so:

```
...
<q-select
        filled
        v-model="model"
        use-input
        input-debounce="0"
        label="Simple filter"
        :options="options"
        @filter="filterFn"
        style="width: 250px"
        data-cy="q-select-test"
>
...
```

Then I opened up the developer tools and inspected the `q-select` element to which I made the changes just described and I see the following:

![image](https://user-images.githubusercontent.com/30523269/142205038-c3d6319e-1109-488d-bc95-9d6768c15824.png)

As you can see the `data-cy` prop is applied to both the input and the div containing the input. This behavior is IMHO not intentional when trying to perform e2e tests using cypress. As I mentioned there are workarounds by e.g., looking only at inputs which have the `data-cy` tag. But considering that the `data-cy` tag should be treated like an id I think it is more obvious if they are uniquely defined which is not possible using `q-select`.

Thanks for any feedback.

**Comment by Quasarman** [comment]
I also had a look at your testing branch and the same thing I described above can be observed without doing any changes to your codepen: https://codepen.io/pdanpdan/pen/GRvPjeK?editors=101

![image](https://user-images.githubusercontent.com/30523269/142206070-6bb8b3fb-c6e5-4178-a441-0805529de2b0.png)

**Comment by pdanpdan** [comment]
Yes, I'll try and look on the code, but there is no way I can think of to differentiate the attrs for wrapper div and for input

**Comment by Quasarman** [comment]
Alright. In the mean time for anyone who might also have this issue I simply defined a separate command for dealing with this case like so:

```
Cypress.Commands.add(
  'dataCyInput',
  {
    prevSubject: 'optional',
  },
  (subject, value) => {
    return cy.get(`input[data-cy=${value}]`, {
      withinSubject: subject,
    });
  },
);
```

This will only specifically look at `data-cy` properties that are applied to input element inside the `q-select` and thus ignore the `data-cy` property applied to the wrapper div.
